src/systems/particles.rs

```diff
@@ -0,0 +1,53 @@
+use crate::{Particle, ParticleEmitter, Velocity};
+use bevy::prelude::*;
+use probability::distribution::Gaussian;
+use probability::prelude::*;
+use rand::distributions::{Bernoulli, Distribution};
+use std::time::Duration;
+
+/// emit new particles from entites with the `ParticleEmitter` component
+pub fn particle_emission_system(
+    mut commands: Commands,
+    query: Query<(&ParticleEmitter, &Transform, &Sprite, &Velocity)>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+) {
+    for (_particle_emmiter, transform, sprite, velocity) in query.iter() {
+        let d = Bernoulli::new(0.3).unwrap();
+        if d.sample(&mut rand::thread_rng()) {
+            let mut source = source::default();
```

Comment:
Where does `source` come from? `rand`?

Reply:
The probability crate. (From the `prelude::*`)

I'm not super happy with this implementation. If you know a better api for sampling from a Gaussian, would be down to rewrite

Reply:
I definitely don't know one, but I was reading this line and got confused. of course it's from the prelude, sometimes I dislike that feature.

Reply:
I usually dislike it. I want it to be explicit where everything comes from.

I did find that when I used Amethyst it helped because there were so many modules, it was impractical otherwise. But that was part of the failure of Amethyst... (too much stuff exposed to the developer to be productive...)